OpenRA.Mods.Common/Scripting/Global/LightingGlobal.cs

```diff
@@ -0,0 +1,66 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using Eluant;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Scripting;
+
+namespace OpenRA.Mods.Common.Scripting
+{
+	[ScriptGlobal("Lighting")]
+	public class LightingGlobal : ScriptGlobal
+	{
+		readonly IEnumerable<FlashPaletteEffect> flashPaletteEffects;
+		readonly GlobalLightingPaletteEffect lighting;
+
+		public LightingGlobal(ScriptContext context)
+			: base(context)
+		{
+			flashPaletteEffects = context.World.WorldActor.TraitsImplementing<FlashPaletteEffect>();
+			lighting = context.World.WorldActor.TraitOrDefault<GlobalLightingPaletteEffect>();
+
+			if (lighting == null)
+				throw new LuaException("World actor must have GlobalLightingPaletteEffect to use the Lighting table.");
```

Comment:
Hmm checking for null (or a bool) in each property access is a bit distasteful, but okay.